plugins/fzuse.rb

```diff
@@ -0,0 +1,150 @@
+        'hl+' => 'red:regular:bold',
+        'info' => '-1',
+        'marker' => '-1',
+        'prompt' => '-1',
+        'spinner' => '-1',
+        'pointer' => 'blue:bold',
+        'header' => '-1',
+        'border' => '-1',
+        'label' => '-1',
+        'query' => '-1'
+      }.freeze
+
+      def initialize(driver)
+        super
+
+        @module_dispatcher = Msf::Ui::Console::CommandDispatcher::Modules.new(driver)
+      end
+
+      def name
+        'FuzzyUse'
+      end
+
+      #
+      # Returns the hash of commands supported by this dispatcher.
+      #
+      def commands
+        {
+          'fzuse' => 'A fuzzy use command added by the FuzzyUse plugin'
+        }
+      end
+
+      def start_pipe_server(socket_path)
+        def pipe_server(socket_path)
+          server = UNIXServer.new(socket_path)
+          File.chmod(0600, socket_path)
+          loop do
+            client = server.accept
+            begin
+              unless (input_string = client.gets&.chomp).blank?
+                if (mod = framework.modules.create(input_string))
+                  client.puts(Serializer::ReadableText.dump_module(mod))
+                end
+              end
+            rescue StandardError
+            end
+            client.close
+          end
+        rescue EOFError
+        ensure
+          server.close if server
+          File.delete(socket_path) if File.exist?(socket_path)
+        end
+
+        Thread.new do
+          pipe_server(socket_path)
+        end
+      end
+
+      #
+      # This method handles the fzuse command.
+      #
+      def cmd_fzuse(*args)
+        selection = nil
+
+        Dir.mktmpdir('msf-fzuse-') do |dir|
+          socket_path = File.join(dir, "msf-fzuse.sock")
+          server_thread = start_pipe_server(socket_path)
+
+          query = args.empty? ? '' : args.first
+          ruby = RbConfig::CONFIG['bindir'] + '/' + RbConfig::CONFIG['ruby_install_name'] + RbConfig::CONFIG['EXEEXT']
+
+          color = "--color=#{FZF_THEME.map { |key, value| "#{key}:#{value}" }.join(',')}"
+          Open3.popen3('fzf', '--select-1', '--query', query, '--pointer=->', color, '--preview', "'#{ruby}' '#{__FILE__}' '#{socket_path}' '{1}'", '--preview-label', "Module Information") do |stdin, stdout, stderr, wait_thr|
+            framework.modules.module_types.each do |module_type|
+              framework.modules.module_names(module_type).each do |module_name|
+                stdin.puts "#{module_type}/#{module_name}"
+              end
+            end
+            stdin.close
+            selection = stdout.read
+          end
+
+          server_thread.kill
+          server_thread.join
+        end
+
+        return if selection.blank?
+
+        selection.strip!
+        @module_dispatcher.cmd_use(selection)
+      end
+    end
+
+    def initialize(framework, opts)
+      super
+
+      missing_requirements = []
+      missing_requirements << 'fzf' unless Msf::Util::Helper.which('fzf')
```

Comment:
I assume a requirement will be running on a unix host, for the UNIXServer requirement